File: leetcode/lc-us/cpp/1526.cpp
#include <bits/stdc++.h>

using namespace std;

class Solution {
public:
    int minNumberOperations(vector<int> &target) {
        int ans = target[0];
        for (int i = 0; i < (int) target.size() - 1; i++) {
            if (target[i + 1] > target[i]) {
                ans += target[i + 1] - target[i];
            }
        }
        return ans;
    }
};

File: leetcode/lc-us/rust/1526.rs
impl Solution {
    pub fn min_number_operations(target: Vec<i32>) -> i32 {
        let mut res = target[0];
        for i in 1..target.len() {
            if target[i] > target[i - 1] {
                res += target[i] - target[i - 1];
            }
        }
        res
    }
}